lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -1360,23 +1360,53 @@ def setUp(self):
         super(TestRebindModule, self).setUp()
         self.homework = self.add_graded_section_to_course('homework')
         self.lti = ItemFactory.create(category='lti', parent=self.homework)
+        self.problem = ItemFactory.create(category='problem', parent=self.homework)
         self.user = UserFactory.create()
         self.anon_user = AnonymousUser()
 
-    def get_module_for_user(self, user):
+    def get_module_for_user(self, user, item=None):
         """Helper function to get useful module at self.location in self.course_id for user"""
         mock_request = MagicMock()
         mock_request.user = user
         field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
             self.course.id, user, self.course, depth=2)
 
+        if item is None:
+            item = self.lti
+
         return render.get_module(  # pylint: disable=protected-access
             user,
             mock_request,
-            self.lti.location,
+            item.location,
             field_data_cache,
         )._xmodule
 
+    def test_rebind_module_to_new_users(self):
+        module = self.get_module_for_user(self.user, self.problem)
+        # read a mutable field, scoped to a single user,
+        # in order to mark it as dirty
+        module.correct_map
+
+        correct_map_field = module.fields['correct_map']
+        self.assertIn(correct_map_field, module._dirty_fields)
+        self.assertIn(correct_map_field.name, module._field_data_cache)
+
+        # Bind the module to another student, which will remove "correct_map"
+        # from the module's _field_data_cache and _dirty_fields.
+        user2 = UserFactory.create()
+        module.bind_for_student(module.system, user2.id)
+        self.assertNotIn(correct_map_field, module._dirty_fields)
+        self.assertNotIn(correct_map_field.name, module._field_data_cache)
```

Comment:
Any way you can describe this behavior in a way that doesn't require a reference to `_dirty_fields` and `_field_data_cache`? What's the expected user-visible behavior?